### ISSUE TITLE
`import-db` fails if gzip file has trailing garbage

### ISSUE THREAD

**Issue by mbomb007** [drud/ddev#3500]
**Run a Diagnostic and Paste Link Here**

https://gist.github.com/mbomb007/4e1424a484e6fa7105dca849f78b0afa

Yes, I know I'm a couple versions behind, however, that's because I need to use xdebug, and that wasn't working after upgrading. However, I checked and nothing was in the changelog regarding `import-db`.

**Describe the bug**

Drupal Backup and Migrate module sometimes creates gzip files with trailing garbage, (or an invalid header?) If I use 7-zip or `gunzip` , they will unzip the file anyway, and show a warning. However, when using `import-db`, it fails with an error:

> Failed to import database db for PROJECT: failed to extract provided archive: gzip: invalid header

So I need to use `gunzip *.gz && ddev import-db < *.mysql`

**To Reproduce**

Try importing a DB backup with trailing garbage in the gzip.

To test whether a gzip has that, run `gzip -t`

> gzip -t backup-2022-01-01T10-04-18.mysql.gz 
> gzip: backup-2022-01-01T10-04-18.mysql.gz: decompression OK, trailing garbage ignored

**Expected behavior**

The file should be unzipped and imported successfully, despite it not conforming to standards.

**Version and configuration information (please complete the following information):**

- Host computer OS and Version: Windows 10 (20H2), WSL 2 (Ubuntu 20.04)
- Docker Desktop version: N/A, I installed on Linux
- ddev version information (use `ddev version`): v1.18.0
- config.yaml contents for the misbehaving project (the part before the comments)

```yaml
name: PROJECT
type: drupal9
docroot: web
php_version: "8.0"
webserver_type: nginx-fpm
router_http_port: "80"
router_https_port: "443"
xdebug_enabled: false
additional_hostnames: []
mariadb_version: "10.5"
mysql_version: ""
nfs_mount_enabled: false
mutagen_enabled: false
hooks:
  post-start:
  - composer: install -d /var/www/html
omit_containers: [dba]
webimage_extra_packages: [build-essential, nodejs, ruby-full, rubygems, freetds-bin,
  freetds-dev, php-sybase, php-snmp, php-apcu, php-gd, php-uploadprogress]
use_dns_when_possible: true
composer_version: "2"
web_environment:
- max_input_vars=3000
```

- Do you have any custom configuration (nginx, php, mysql) in the .ddev folder? Not anything relevant

**Additional context**

**Comment by rfay** [comment]
Doesn't it seem like you should file the issue on the creator of the invalid file?

Please look at your file using `zless` and see what is the crap at the end and that will probably help you understand it. 

gunzip doesn't like invalid files :)

If you need help with your xdebug, please ask. It's working for everybody else. I always want to hear about problems people are having.

**Comment by mbomb007** [comment]
When using `zless` to inspect the .mysql.gz file, then pressing Ctrl+End (and waiting a very long time), it shows this:

```
/*!40101 SET SQL_MODE=@OLD_SQL_MODE */;
/*!40014 SET FOREIGN_KEY_CHECKS=@OLD_FOREIGN_KEY_CHECKS */;
/*!40014 SET UNIQUE_CHECKS=@OLD_UNIQUE_CHECKS */;
/*!40101 SET CHARACTER_SET_RESULTS=@OLD_CHARACTER_SET_RESULTS */;
/*!40101 SET COLLATION_CONNECTION=@OLD_COLLATION_CONNECTION */;
/*!40101 SET CHARACTER_SET_CLIENT=@OLD_CHARACTER_SET_CLIENT */;

SET FOREIGN_KEY_CHECKS = 1;
COMMIT;
SET AUTOCOMMIT = 1;The website encountered an unexpected error. Please try again later.
```

If I extract using 7-Zip and go to the end of the .mysql file, I see the same, but without the error message. So the garbage must be `The website encountered an unexpected error. Please try again later.`.

**Comment by rfay** [comment]
Yes, that's an invalid file, and you will want to work on ways to get a valid db dump.

I assume this is from a particular procedure that you're using to get a dump. You'll want to work on that procedure, because you actually have a corrupted dump. Let me know if I can help with that.

**Comment by rfay** [comment]
I'm going to close this, but happy to continue the conversation here or elsewhere, and happy to help figure out what's going on with your db dump.

**Comment by Schrank** [comment]
In case anyone is finding this issue as I did: In my case the problem was (don't ask me why), that the file was `dump.gz`, renaming it to `dump.sql.gz` made it work 🤷

**Comment by rfay** [comment]
Thanks @Schrank - DDEV determines what to do with a file based on its filiename, and sql.gz and mysql.gz are the ones that are expected to expand to a plain .sql file.

**Comment by JPustkuchen** [comment]
In case you're trying to import a dump created by backup_migrate 3.x in Drupal 7, you might be affected by this issue: https://www.drupal.org/project/backup_migrate/issues/2946032

TL;DR:
The file has a .mysql.gz suffix, but in fact is not compressed and should be named ".mysql". "GZip" compression is enabled in the backup_migrate profile settings!
